### PR TITLE
feat: implement aggregate func compact_state_agg

### PR DESCRIPTION
**Pull request by yukkit**
# Required checklist
- [x] Sample config files updated (`config`,`meta/config` and `default config`) 
- [x] If there are user-facing changes, the documentation needs to be updated prior to approving the PR( [Link]() )
- [x] If there are any breaking changes to public APIs, please add the `api change` label.
- [x] Signed [CLA](https://cla-assistant.io/cnosdb/cnosdb) (if not already signed)

# Which issue does this PR close?

[//]: # (We generally require a GitHub issue to be filed for all bug fixes and enhancements and this helps us generate change logs for our releases. You can link an issue to this PR using the GitHub syntax. For -- example `Closes #123` indicates that this PR will close issue #123.)

Related #1358 .

# Rationale for this change

[//]: # (Why are you proposing this change? If this is already explained clearly in the issue then this section is not needed.
 Explaining clearly why changes are proposed helps reviewers understand your changes and offer better suggestions for fixes.)

# Are there any user-facing changes?

[//]: # (There is no need to duplicate the description in the issue here, but it is sometimes worth providing a summary of the individual changes in this PR.)

### CODE REVIEW

File: query_server/query/src/extension/expr/aggregate_function/state_agg/compact_state_agg.rs [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+        let time_data_type = self.time_records[0].get_datatype();
+        let state_data_type = self.state_value_records[0].get_datatype();
+
+        let time_state = ScalarValue::new_list(Some(self.time_records.clone()), time_data_type);
+        let value_state =
+            ScalarValue::new_list(Some(self.state_value_records.clone()), state_data_type);
+
+        trace::trace!("time_state data type: {:?}", time_state.get_datatype());
+
+        time_state.to_array();
```
Comment (zipper-meng) [inline query_server/query/src/extension/expr/aggregate_function/state_agg/compact_state_agg.rs:318]: Why was this 1-row array thrown away?

**Review by zipper-meng — APPROVED** [review]
It's fine.

File: query_server/query/src/extension/expr/aggregate_function/state_agg/compact_state_agg.rs (hunk 1 above)
Comment (yukkit) [reply query_server/query/src/extension/expr/aggregate_function/state_agg/compact_state_agg.rs:318]: This line should be deleted

**Review by zipper-meng — APPROVED** [review]
It's fine, two.

**Pull request merged**